tests/test_freqtradebot.py

```diff
@@ -3382,7 +3392,8 @@ def test_execute_trade_exit_with_stoploss_on_exchange(
         price_to_precision=lambda s, x, y: y,
         stoploss=stoploss,
         cancel_stoploss_order=cancel_order,
-        _is_dry_limit_order_filled=MagicMock(side_effect=[True, False]),
+        _fill_dry_limit_order=MagicMock(
+            side_effect=SideEffect(lambda *_: _[-2:], lambda *_: (None, 0))),
```

Comment:
can't this also be done by using an array passed to side_effect?

I know we use this in other places - but most likely not with functions.

Reply:
[side-effect is overloaded to serve a different purpose for iterables.](https://stackoverflow.com/questions/64992044/pytest-mock-multiple-calls-of-same-method-with-different-side-effect/64992350#:~:text=side-effect%20is%20overloaded%20to%20serve%20a%20different%20purpose%20for%20iterables)